### PR TITLE
Add Conv3D gradient test and vjp fixes

### PR DESCRIPTION
**Pull request by Shashi456**
I think I was the one who originally wrote the layer, Sorry for the errors. 
Build and test pass locally.

### CODE REVIEW

File: Sources/TensorFlow/Operators/NN.swift [hunk 1]
```diff
@@ -268,11 +266,9 @@ func _vjpConv3DBackpropInput<Scalar: TensorFlowFloatingPoint>(
     let value = conv3DBackpropInput(x, shape: shape, filter: filter, strides: strides,
                                     padding: padding)
     return (value, { v in
-        return (
-            conv3DBackpropFilter(x, input: v, filterSizes: shape, strides: strides,
-                                 padding: padding),
-            conv3D(v, filter: filter, strides: strides, padding: padding)
-        )
+         (conv3D(v, filter: filter, strides: strides, padding: padding),
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]: Indent by 4, not 5.
```suggestion
        (conv3D(v, filter: filter, strides: strides, padding: padding),
```

File: Sources/TensorFlow/Operators/NN.swift [hunk 2]
```diff
@@ -268,11 +266,9 @@ func _vjpConv3DBackpropInput<Scalar: TensorFlowFloatingPoint>(
...
+          conv3DBackpropFilter(x, input: v, filterSizes: filter.shapeTensor, strides: strides,
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]:
```suggestion
         conv3DBackpropFilter(x, input: v, filterSizes: filter.shapeTensor, strides: strides,
```

File: Sources/TensorFlow/Operators/NN.swift [hunk 3]
```diff
@@ -268,11 +266,9 @@ func _vjpConv3DBackpropInput<Scalar: TensorFlowFloatingPoint>(
...
+                               padding: padding))
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]:
```suggestion
                              padding: padding))
```

File: Sources/TensorFlow/Operators/NN.swift [hunk 4]
```diff
@@ -306,11 +302,9 @@ func _vjpConv3DBackpropFilter<Scalar: TensorFlowFloatingPoint>(
     let value = conv3DBackpropFilter(x, input: input, filterSizes: filterSizes,
                                      strides: strides, padding: padding)
     return (value, { v in
-        return (
-            conv3DBackpropInput(x, shape: filterSizes, filter: v, strides: strides,
-                                  padding: padding),
-            conv3D(input, filter: v, strides: strides, padding: padding)
-        )
+         (conv3D(input, filter: v, strides: strides, padding: padding),
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]:
```suggestion
        (conv3D(input, filter: v, strides: strides, padding: padding),
```

File: Sources/TensorFlow/Operators/NN.swift [hunk 5]
```diff
@@ -306,11 +302,9 @@ func _vjpConv3DBackpropFilter<Scalar: TensorFlowFloatingPoint>(
...
+          conv3DBackpropInput(x, shape: x.shapeTensor, filter: v, strides: strides,
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]:
```suggestion
         conv3DBackpropInput(x, shape: x.shapeTensor, filter: v, strides: strides,
```

File: Sources/TensorFlow/Operators/NN.swift [hunk 6]
```diff
@@ -306,11 +302,9 @@ func _vjpConv3DBackpropFilter<Scalar: TensorFlowFloatingPoint>(
...
+                              padding: padding))
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]:
```suggestion
                             padding: padding))
```

**Comment by Shashi456** [comment]
@rxwei could you put a build on this again, since we removed the failing test.

**Comment by Shashi456** [comment]
can this and the tranposed conv PRs be merged?

**Comment by Shashi456** [comment]
I'll change this wrt #499 and get this up in a while.

**Comment by Shashi456** [comment]
@rxwei done.

**Comment by Shashi456** [comment]
@rxwei could you trigger a build on this again

File: Tests/TensorFlowTests/LayerTests.swift [hunk 7]
```diff
@@ -197,6 +197,55 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output, expected)
     }
 
+    func testConv3DGradient() {
+        let filter =  Tensor(shape: [1, 4, 4, 1, 1], scalars: (0..<16).map(Float.init))
+        let bias = Tensor<Float>(ones: [2])
+        let layer = Conv3D(filter: filter,
+                            bias: bias,
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]: Fix indentation.

**Comment by Shashi456** [comment]
@rxwei fixed it :)

**Comment by Shashi456** [comment]
@rxwei this can be merged.

File: Tests/TensorFlowTests/LayerTests.swift [hunk 8]
```diff
@@ -197,6 +197,55 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output, expected)
     }
 
+    func testConv3DGradient() {
+        let filter =  Tensor(shape: [1, 4, 4, 1, 1], scalars: (0..<16).map(Float.init))
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]: Please make sure to use proper spacing or run swift-format before pushing.

```suggestion
        let filter = Tensor(shape: [1, 4, 4, 1, 1], scalars: (0..<16).map(Float.init))
```

File: Tests/TensorFlowTests/LayerTests.swift [hunk 9]
```diff
@@ -197,6 +197,55 @@ final class LayerTests: XCTestCase {
...
+        let bias = Tensor<Float>(ones: [2])
+        let layer = Conv3D(filter: filter,
+                           bias: bias,
+                           activation: identity,
+                           strides: (2, 2, 2),
+                           padding: .same)
+        let input = Tensor(shape: [1, 4, 4, 4, 1], scalars: (0..<64).map(Float.init))
+        let grads = gradient( at: input, layer) { $1($0).sum() }
```
Comment (rxwei) [inline Tests/TensorFlowTests/LayerTests.swift]:
```suggestion
        let grads = gradient(at: input, layer) { $1($0).sum() }
```

**Pull request merged**